Integrations/Forescout/Forescout.py

```diff
@@ -0,0 +1,678 @@
+        }
+        included_fields_readable = {
+            HOSTFIELDS_TO_INCLUDE.get(key): dict_to_formatted_string(val)
+            for key, val in fields.items() if key in HOSTFIELDS_TO_INCLUDE.keys()
+        }
+    elif requested_fields:
+        included_fields = fields
+        included_fields_readable = {key: dict_to_formatted_string(val) for key, val in fields.items()}
+
+    content = {
+        'ID': host.get('id'),
+        'IP': host.get('ip'),
+        'MAC': host.get('mac'),
+        'EndpointURL': data.get('_links', {}).get('self', {}).get('href'),
+        'Field': included_fields
+    }
+
+    human_readable_content = deepcopy(content)
+    human_readable_content['Field'] = included_fields_readable
+
+    context = {'Forescout.Host(val.ID && val.ID === obj.ID)': content}
+
+    title = 'Endpoint Details for {}'.format(identifier) if identifier else 'Endpoint Details'
+    human_readable = tableToMarkdown(title, human_readable_content, removeNull=True)
+    return_outputs(readable_output=human_readable, outputs=context, raw_response=data)
+
+
+def get_hosts(args={}):
+    url_suffix = '/api/hosts'
+    headers = create_web_api_headers()
+    params: Dict = {}
+    rule_ids = args.get('rule_ids')
+    properties = args.get('properties')
+    if rule_ids and properties:
+        url_suffix += '?matchRuleId=' + rule_ids + '&' + properties
+    elif rule_ids:
+        url_suffix += '?matchRuleId=' + rule_ids
+    elif properties:
+        url_suffix += '?' + properties
+    response = http_request('GET', url_suffix, headers=headers, params=params, resp_type='response')
+    return response
+
+
+def get_hosts_command():
+    args = demisto.args()
+    response = get_hosts(args).json()
+    content = [
+        {
+            'ID': x.get('hostId'),
+            'IP': x.get('ip'),
+            'MAC': x.get('mac'),
+            'EndpointURL': x.get('_links', {}).get('self', {}).get('href')
+        } for x in response.get('hosts', [])
+    ]
+    context = {'Forescout.Host(val.ID && val.ID === obj.ID)': content}
+    title = 'Active Endpoints'
+    human_readable = tableToMarkdown(title, content, removeNull=True)
+    return_outputs(readable_output=human_readable, outputs=context, raw_response=response)
+
+
+def get_hostfields():
+    url_suffix = '/api/hostfields'
+    headers = create_web_api_headers()
+    response = http_request('GET', url_suffix, headers=headers, resp_type='response')
+    return response
+
+
+def get_hostfields_command():
+    args = demisto.args()
+    response = get_hostfields()
+    data = response.json()
+    filtered_data = filter_hostfields_data(args, data)
+    if not filtered_data:
+        demisto.results('No hostfields matched the specified filters.')
+    else:
+        content = [{key.title(): val for key, val in x.items()} for x in filtered_data]
+        context = {'Forescout.HostField': content}
```

Comment:
no unique field here ?

Reply:
there are over 900 potential fields that can be returned...